src/jit/codegenarmarch.cpp

```diff
@@ -781,14 +781,12 @@ void CodeGen::genCodeForArrOffset(GenTreeArrOffs* arrOffset)
         var_types elemType = arrOffset->gtArrElemType;
         unsigned  offset   = genOffsetOfMDArrayDimensionSize(elemType, rank, dim);
 
-// Load tmpReg with the dimension size and evaluate
-// tgtReg = offsetReg*dim_size + indexReg.
+        // Load tmpReg with the dimension size and evaluate
+        // tgtReg = offsetReg*tmpReg + indexReg.
+        emit->emitIns_R_R_I(ins_Load(TYP_INT), EA_PTRSIZE, tmpReg, arrReg, offset);
```

Comment:
If the address in the destination register points at the start of a valid GC object use EA_GCREF,
If the address in the destination registers points at data inside a GC object use EA_BYREF

I think that 'arrReg' is a valid GC object and that anything added to it is a BYREF pointer.